### PR TITLE
Model.compile() requires a loss argument

### PR DESCRIPTION
**Pull request by larsr**
The change in [`examples/variational_autoencoder.py`](https://github.com/keras-team/keras/blob/master/examples/variational_autoencoder.py) was introduced in

https://github.com/keras-team/keras/commit/10b46cbdc47f7164cd92209077939a7f7882b3c0

It is missing a `loss` argument.

Here is where Model.compile() is defined:

https://github.com/keras-team/keras/blob/d9f26a92f4fdc1f1e170a4203a7c13abf3af86e8/keras/models.py#L758

### CODE REVIEW

**Comment by fchollet** [comment]
The `loss` argument in compile is optional (it defaults to None). Losses may be set via other means, such as calls to `add_loss` (like in this example), in which case an external loss function is not required to be applied to model outputs.
